### PR TITLE
support testing when training and handle dropout and batch_norm operator in testing mode

### LINKED ISSUE #5733: need to handle dropout/batch_norm operator in testing mode
Dropout and batch norm operator do different between training and testing mode. We should handle these operators when users inference data after training.

### LINKED ISSUE #5814: Support testing while training

### PR DESCRIPTION
**Pull request by QiJune**
Fix #5733 and Fix #5814

### CODE REVIEW

File: paddle/framework/prune.cc [hunk 1]
```diff
@@ -99,11 +102,23 @@ void prune_impl(const ProgramDesc& input, ProgramDesc* output, int block_id) {
       *op_field->Add() = input.blocks(block_id).ops(i);
     }
   }
+  if (is_test) {
+    for (auto& op_desc : *op_field) {
+      if (op_desc.type() == kDropOutOpType ||
+          op_desc.type() == kBatchNormOpType) {
+        for (auto& attr : *op_desc.mutable_attrs()) {
+          if (attr.name() == "is_test") {
+            attr.set_b(true);
```
Comment (Canpio) [inline paddle/framework/prune.cc]: `break;`

File: paddle/framework/prune.cc [hunk 2]
```diff
@@ -99,11 +102,23 @@ void prune_impl(const ProgramDesc& input, ProgramDesc* output, int block_id) {
...
+      }
+    }
+  }
 }
 
 // TODO(fengjiayi): Prune() could be inplaced to avoid unnecessary copies
-void Prune(const ProgramDesc& input, ProgramDesc* output) {
-  prune_impl(input, output, 0);
+void Prune(const ProgramDesc& input, ProgramDesc* output, bool is_test) {
+  prune_impl(input, output, 0, is_test);
```
Comment (Canpio) [inline paddle/framework/prune.cc]: I think setting attribute `is_test` true has no relationship with `prune`. We shall add an independent function or module to do this.

Comment (QiJune) [reply paddle/framework/prune.cc]: We now have `prune` method to transform ProgramDesc and generate another ProgramDesc.
Exactly, we will do some optimization in inference stage and handle some special operators. Then, we will generate a inference ProgramDesc.
Maybe we need another method like `prune`, the interface could be:
```
void InferenceOptimize(const ProgramDesc& input, ProgramDesc* output);
```

File: paddle/framework/prune.cc (hunk 1 above)
Comment (QiJune) [reply paddle/framework/prune.cc]: Done

File: paddle/framework/prune.cc [hunk 3]
```diff
@@ -106,5 +108,26 @@ void Prune(const ProgramDesc& input, ProgramDesc* output) {
   prune_impl(input, output, 0);
 }
 
+void inference_optimize_impl(const ProgramDesc& input, ProgramDesc* output,
```
Comment (reyoung) [inline paddle/framework/prune.cc]: The logic of `inference_optimize_impl` is quite simple. Maybe we can implement it in Python.
